File: catapult/python/block_info.py
from datetime import datetime

import httpx
from helper import get_config


def block_info():
    config = get_config()
    url = config["private1"]["url"]

    network = httpx.get(url + "/network/properties").json()
    epoch = network["network"]["epochAdjustment"]
    epoch = epoch.replace("s", "")

    params = {"pageNumber": 1, "pageSize": 1, "order": "desc"}
    block_info = httpx.get(url + "/blocks", params=params).json()

    for data in block_info["data"]:
        meta = data["meta"]
        block = data["block"]
        block_timestamp = int(block["timestamp"]) / 1000
        print("-------------------------------------------------")
        print("BlockHeight", block["height"])
        print("Date", datetime.fromtimestamp(block_timestamp + int(epoch)))
        print("TotalTransaction", meta["transactionsCount"])
        print("-------------------------------------------------")


if __name__ == "__main__":
    block_info()

File: catapult/javascript/block_info.js
const { RepositoryFactoryHttp, Order } = require("@tech-bureau/symbol-sdk");
const config = require("config");
const url = config.get("private1.url");

const { firstValueFrom } = require("rxjs");

(async () => {
  const repo = new RepositoryFactoryHttp(url);

  //Networktypeをcatapultから取得
  const networkType = await firstValueFrom(repo.getNetworkType());
  //GenerationHashをcatapultから取得
  const ghash = await firstValueFrom(repo.getGenerationHash());
  //GenerationHashをcatapultから取得
  const epoch = await firstValueFrom(repo.getEpochAdjustment());

  console.log("network", networkType);
  console.log("GenerationHash", ghash);

  try {
    const blockRepo = repo.createBlockRepository();

    const searchCriteria = {
      pageNumber: 1,
      pageSize: 10,
      order: Order.Desc,
    };

    const blockInfo = await firstValueFrom(blockRepo.search(searchCriteria));

    blockInfo.data.map((block) => {
      console.log("-------------------------------------------------");
      console.log("BlockHeight", block.height.compact());
      console.log("Date", new Date(block.timestamp.compact() + epoch * 1000).toString());
      console.log("TotalTranction", block.transactionsCount);
      // console.log('BlockInfo', block)
      console.log("-------------------------------------------------");
    });
  } catch (e) {
    console.error(e);
  }
})();
